audits/v3/Audit2.md

```diff
@@ -0,0 +1,80 @@
+| Checked    | `DePayRouterV3.sol`                         |  
+| Branch     | v3                                          |  
+| Time       | 2024-09-30                                  |  
+| Author     | Nikolai Mikhailov                           |  
+
+# Result
+
+| Severity        | Count | Link                                                                                             |  
+| :-------------: | ----: | ------------------------------------------------------------------------------------------------ |  
+| High            | 0     |                                                                                                  |  
+| Medium          | 1     |                                                                                                  |  
+|                 |       | [M01 - ProtocolAmount is not enforced, allowing users to bypass protocol fees](#m01)             |  
+| Low             | 0     |                                                                                                  |  
+|                 |       |                                                                                                  |  
+| Informational   | 1     |                                                                                                  |  
+|                 |       | [I01 - Inconsistent time units used for deadlines (milliseconds vs seconds)](#i02)               |  
+
+---
+
+<a name="m01"></a>
+
+## M01 - ProtocolAmount is not enforced, allowing users to bypass protocol fees
+
+| Affected            | Severity | Count | Lines                                                              |  
+| :---------------:   | :------: | ----: | ------------------------------------------------------------------ |  
+| `DePayRouterV3.sol` | Low      | 1     | Functions handling protocolAmount in payment processing            |  
+
+**Description**
+
+The contract intends to collect a protocol fee specified by `protocolAmount` within the Payment struct. However, this amount is provided by the user and not calculated or enforced by the contract. Users can set `protocolAmount` to zero or any arbitrary value, effectively bypassing the protocol fee mechanism.
+
+**Impact**
+
+The lack of enforcement allows users to avoid paying protocol fees, leading to potential revenue loss for the protocol and undermining the intended fee structure.
+
+**Recommendation**
+
+- Calculate Protocol Fee Internally: Remove protocolAmount from the user-supplied Payment struct. Instead, calculate it within the contract based on a predefined fee percentage of the transaction amount (e.g., paymentAmount or amountIn).
+
+- Enforce Protocol Fee: Ensure that the calculated protocolAmount is correctly retained within the contract after the transaction and is not transferred to the payment receiver or any other party.
+
+---
+
+<a name="i01"></a>
+
+## I01 - Inconsistent time units used for deadlines (milliseconds vs seconds)
```

Comment:
The use of `deadline` in milliseconds is explicitly documented both in the source code and accompanying documentation. The requirement for `deadline` in milliseconds serves a specific purpose: it allows for the creation of a more unique index that includes the payment sender (`from`), payment receiver (`to`), and the `deadline`. This ensures that payments made within the same second are distinguishable, providing a unique index for validation and preventing duplicate payments in time-sensitive scenarios.